src/service/gif.js

```diff
@@ -1,9 +1,13 @@
 import api from './config';
 
-export const getGifFeatured = async () =>
+export const getGifFeatured = async (search) =>
   new Promise((resolve, reject) => {
     api
-      .get('/chat/gif/featured')
+      .get(
+        search?.trim() !== ''
+          ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
+          : '/chat/gif/featured'
+      )
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `getGifFeatured` function is good as it now supports searching for gifs. However, it's important to handle potential errors that might occur during the API call. Currently, if the API request fails, the Promise will be left hanging because there's no error handling in place. You can add a `.catch` block to handle any errors that might occur.
```diff
 import api from './config';
 
export const getGifFeatured = async (search) =>
   new Promise((resolve, reject) => {
     api
      .get(
        search?.trim() !== ''
          ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
          : '/chat/gif/featured'
      )
      .then((res) => {
         resolve(res.data);
       })
+      .catch((error) => {
+        reject(error);
+      });
   });
```
This way, if an error occurs during the API call, it will be passed to the `reject` function and can be handled by the caller of `getGifFeatured`.


<!-- This is an auto-generated comment by OSS CodeRabbit -->